### ISSUE TITLE
FAILED: libqemuutil.a.p/util_async.c.obj on windows + mingw64

### ISSUE THREAD

**Issue by dyharlan** [devos50/qemu#94]
I keep getting these errors on mingw64 on the ipod 2g branch after following the instructions:

```
[644/2160] Compiling C object libqemuutil.a.p/util_async.c.obj
FAILED: libqemuutil.a.p/util_async.c.obj
"cc" "-m64" "-mcx16" "-Ilibqemuutil.a.p" "-I." "-I.." "-Iqapi" "-Itrace" "-Iui" "-Iui/shader" "-ID:/Programs/MSYS64/mingw64/include/glib-2.0" "-ID:/Programs/MSYS64/mingw64/lib/glib-2.0/include" "-ID:/Programs/MSYS64/mingw64/include/pixman-1" "-I/mingw64/include/openssl" "-fdiagnostics-color=auto" "-Wall" "-Winvalid-pch" "-Werror" "-std=gnu11" "-O2" "-g" "-iquote" "." "-iquote" "D:/Others/src/qemu-ios" "-iquote" "D:/Others/src/qemu-ios/include" "-iquote" "D:/Others/src/qemu-ios/tcg/i386" "-U_FORTIFY_SOURCE" "-D_FORTIFY_SOURCE=2" "-fno-pie" "-no-pie" "-D_GNU_SOURCE" "-D_FILE_OFFSET_BITS=64" "-D_LARGEFILE_SOURCE" "-Wstrict-prototypes" "-Wredundant-decls" "-Wundef" "-Wwrite-strings" "-Wmissing-prototypes" "-fno-strict-aliasing" "-fno-common" "-fwrapv" "-Wold-style-declaration" "-Wold-style-definition" "-Wtype-limits" "-Wformat-security" "-Wformat-y2k" "-Winit-self" "-Wignored-qualifiers" "-Wempty-body" "-Wnested-externs" "-Wendif-labels" "-Wexpansion-to-defined" "-Wimplicit-fallthrough=2" "-Wno-missing-include-dirs" "-Wno-shift-negative-value" "-Wno-psabi" "-pthread" -MD -MQ libqemuutil.a.p/util_async.c.obj -MF "libqemuutil.a.p/util_async.c.obj.d" -o libqemuutil.a.p/util_async.c.obj "-c" ../util/async.c
In file included from D:/Others/src/qemu-ios/include/qemu/coroutine.h:18,
                 from D:/Others/src/qemu-ios/include/block/aio.h:20,
                 from ../util/async.c:28:
../util/async.c: In function 'aio_bh_poll':
D:/Others/src/qemu-ios/include/qemu/queue.h:303:22: error: storing the address of local variable 'slice' in '*ctx.bh_slice_list.sqh_last' [-Werror=dangling-pointer=]
  303 |     (head)->sqh_last = &(elm)->field.sqe_next;                          \
      |     ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
../util/async.c:162:5: note: in expansion of macro 'QSIMPLEQ_INSERT_TAIL'
  162 |     QSIMPLEQ_INSERT_TAIL(&ctx->bh_slice_list, &slice, next);
      |     ^~~~~~~~~~~~~~~~~~~~
../util/async.c:157:17: note: 'slice' declared here
  157 |     BHListSlice slice;
      |                 ^~~~~
../util/async.c:155:29: note: 'ctx' declared here
  155 | int aio_bh_poll(AioContext *ctx)
      |                 ~~~~~~~~~~~~^~~
cc1.exe: all warnings being treated as errors
[645/2160] Compiling C object libqemuutil.a.p/util_lockcnt.c.obj
[646/2160] Compiling C object libqemuutil.a.p/util_main-loop.c.obj
[647/2160] Compiling C object libqemuutil.a.p/util_qemu-coroutine.c.obj
ninja: build stopped: subcommand failed.
make: *** [Makefile:165: run-ninja] Error 1
```

**Comment by MrDiagnose** [comment]
same :/

**Comment by devos50** [comment]
> cc1.exe: all warnings being treated as errors

It looks like any warning during compilation will halt the compilation process. I think there should be an option in `mingw64` to disable that behavior, as my code also occasionally raises warnings.

**Comment by ChaseKnowlden** [comment]
Fixed after updating to QEMU 8.2.0

**Comment by xiaokamikami** [comment]
> Fixed after updating to QEMU 8.2.0

Could you please provide the commit for this fix? Thanks
